program/src/instruction.rs

```diff
@@ -204,22 +249,45 @@ impl PaladinGovernanceInstruction {
     pub fn unpack(input: &[u8]) -> Result<Self, ProgramError> {
         match input.split_first() {
             Some((&0, _)) => Ok(Self::CreateProposal),
-            Some((&1, _)) => Ok(Self::CancelProposal),
-            Some((&2, _)) => Ok(Self::BeginVoting),
-            Some((&3, rest)) if rest.len() == 1 => {
+            Some((&1, rest)) => {
+                #[derive(BorshDeserialize)]
+                struct Instruction {
+                    instruction_program_id: Pubkey,
+                    instruction_account_metas: Vec<ProposalAccountMeta>,
+                    instruction_data: Vec<u8>,
+                }
```

Comment:
heh, nice and sneaky way to avoid deriving the borsh traits on the instruction, well done!